lib/store/index.js

```diff
@@ -199,10 +200,11 @@ export class Store {
     this.runningKernels = this.runningKernels.filter(k => k !== kernel);
   }
 
-  getFilesForKernel(kernel: Kernel): Array<?string> {
+  getFilesForKernel(kernel: Kernel): Array<string> {
     const grammar = kernel.grammar.name;
     return this.filePaths.filter(file => {
       const kernelOrMap = this.kernelMapping.get(file);
+      if (!kernelOrMap) return [];
```

Comment:
We want to return boolean values here inside of the filter function. If I'm reading this right, an array returned would be coerced to `true` and this could potentially return an array with an irrelevant file.

Reply:
wait then why the `: Array<string>` is that not the return value in flow

Reply:
You may misunderstand the return value from `filter` function.
Refer to this: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter

Reply:
Oh no I see now. I didnt see the return filter was wrapping all of the other returns.